Data/Primitive/SmallArray.hs

```diff
@@ -111,6 +116,11 @@ newtype SmallArray a = SmallArray (Array a) deriving
   , MonadFix
   , Monoid
   , Typeable
+#if MIN_VERSION_base(4,9,0) || MIN_VERSION_transformers(0,4,0)
+  , Eq1
+  , Ord1
```

Comment:
Perhaps we should derive `Read1` here as well?

Reply:
Your thoroughness knows no bounds.